Add days-in-past to keep the remote calendar clean.

whirm opened the pull request:
----
Also clean up mixed tabs and indentation.

Original "days in past" commit author: Martin Kaffanke <m.kaffanke_at_gmail.com>

After/if this PR gets merged I've got some more changes I've collected from various forks (like vtodo support) cleaned up so they apply on top of this.
#

dengste commented:
Thanks for your work. I'm usually opposed to commits only containing tab/whitespace changes because IMO it's a non-issue and makes git-blame more tedious. Are these really necessary?

vanicat commented:
Problems with indention some time render the code difficult to read. It make this code more difficult to correct and improve that it should be.

dengste commented:
I what way is the code difficult to read? Because of the tabs? If so, then you deliberately changed the defaults for Emacs-Lisp mode. Leave 'tab-width' at '8' and everything will be fine. If you look at the Emacs code base, you will see that there's also a wild mixture of tabs and whitespace. It's really a non-issue unless you change 'tab-width' or otherwise make tabs explicitly visible.

vanicat commented on org-caldav.el:
```diff
@@ -158,6 +158,14 @@ doc-string of `org-agenda-skip-if'.  Any entry that matches will
 not be exported.  Note that the normal `org-agenda-skip-function'
 has no effect on the icalendar exporter.")
 
+(defvar org-caldav-days-in-past nil
```
```suggestion
(defcustom org-caldav-days-in-past nil
```

vanicat commented on org-caldav.el:
```diff
@@ -158,6 +158,14 @@ doc-string of `org-agenda-skip-if'.  Any entry that matches will
+This makes it very easy to keep the remote calendar clean.
+
+nil means include all entries (default)
+any number set will cut the dates older than N days in the past.
```
```suggestion
any number set will cut the dates older than N days in the past."
  :type '(choice (const nil :tag "include all entries")
		 (integer :tag "number of days before today to skip in exported calendar")))
```

Not sure how this multiline suggestion work... that said, as most defvar has been made defcustom, this one should be a defcustom too.

vanicat commented on org-caldav.el:
```diff
@@ -158,6 +158,14 @@ doc-string of `org-agenda-skip-if'.  Any entry that matches will
+
+nil means include all entries (default)
+any number set will cut the dates older than N days in the past.
+")
```
```suggestion
```
To have a full changeset: those ") are part of my previous suggested change.

vanicat commented:
Okay, sorry, I wrote my comment based on memory, and i cannot find in the current code the problem I had in mind. I should have checked before writing my comment.

vanicat commented on org-caldav.el:
```diff
@@ -1189,8 +1197,11 @@ is on s-expression."
+             (ts (when org-caldav-days-in-past (* (abs org-caldav-days-in-past) -1)))
+             (stamp (org-entry-get nil "TIMESTAMP" t)))
+         (when (or pt (and stamp (> ts (org-time-stamp-to-now stamp))))
+           (delete-region (point) (or (org-end-of-subtree t)))))))))
```
Not sure why there is this or. The following code is equivalent, and simpler:
```suggestion
           (delete-region (point) (org-end-of-subtree t))))))))
```

whirm replied:
Good catch, It probably was introduced by me while rebasing.
Thanks for the review!

whirm commented:
I've rebased against current `master` and fixed @vanicat's comments. 
I've removed the white space cleanup commits from this PR for now and I'll see if it makes bringing in other changes not too annoying.
